File: fixtures/scale/java/large/src/cryptobench/ciphers/feistel/FeistelCipher.java
package cryptobench.ciphers.feistel;

import cryptobench.core.Cipher;
import cryptobench.util.Bytes;
import cryptobench.util.Hex;

public class FeistelCipher implements Cipher {
    private final FeistelNetwork network;

    public FeistelCipher(FeistelKey key) {
        this.network = new FeistelNetwork(key);
    }

    @Override
    public String name() { return "feistel"; }

    @Override
    public String encrypt(String plaintext) {
        byte[] data = Bytes.pad(Bytes.of(plaintext), 8);
        for (int off = 0; off < data.length; off += 8) network.block(data, off, false);
        return Hex.encode(data);
    }

    @Override
    public String decrypt(String ciphertext) {
        byte[] data = Hex.decode(ciphertext);
        for (int off = 0; off < data.length; off += 8) network.block(data, off, true);
        return Bytes.toText(data).trim();
    }
}

File: fixtures/scale/dart/large/lib/ciphers/feistel/feistel_cipher.dart
import '../../core/cipher.dart';
import '../../util/bytes.dart';
import '../../util/hex.dart';
import 'feistel_key.dart';
import 'feistel_network.dart';

class FeistelCipher implements Cipher {
  final FeistelNetwork _network;

  FeistelCipher(FeistelKey key) : _network = FeistelNetwork(key);

  @override
  String name() => 'feistel';

  @override
  String encrypt(String plaintext) {
    final data = padBytes(bytesOf(plaintext), 8);
    for (var off = 0; off < data.length; off += 8) {
      _network.block(data, off, false);
    }
    return hexEncode(data);
  }

  @override
  String decrypt(String ciphertext) {
    final data = hexDecode(ciphertext);
    for (var off = 0; off < data.length; off += 8) {
      _network.block(data, off, true);
    }
    return bytesToText(data).trim();
  }
}
